### DISCUSSION TITLE
need help

### DISCUSSION THREAD

**Discussion by utwtjm** [artilleryio/artillery#1289] [Artillery]
my-scenario.yaml

------------

config:
  target: "http://127.0.0.1:3000"
  socketio:
    path: "/socket"
    transports: ["websocket"]
  phases:
    - duration: 60
      arrivalRate: 25

scenarios:
  - name: "emit"
    engine: socketio
    flow:
      - emit:
          channel: "test"
          data: {id: "123"}
          acknowledge:
            match:
              json: "$.status"
              value: 0

------------

acknowledge not work 

------------

code file : artillery/core/lib/engine_socketio.js

function isAcknowledgeRequired(spec) {
  console.log(spec);
  return spec.emit && spec.acknowledge;
}

------------

console.log output 

------------

{
  emit: {
    channel: 'test',
    data: { id: '123' },
    acknowledge: { match: [Object] }
  },
  namespace: ''
}

spec.acknowledge not get right value

thank you

**Comment by hassy** [comment]
hi @utwtjm, what do you expect the value to be?

**Reply by utwtjm** [reply]
hi @hassy expect the value is 

{
emit: {
  channel: 'test',
  data: { id: '123' },
},
namespace: ''
acknowledge: { match: [Object] }
}

because function isAcknowledgeRequired 
check --> spec.emit && spec.acknowledge;

thank you
